src/chainstate/stacks/miner.rs

```diff
@@ -1139,34 +1123,30 @@ impl StacksBlockBuilder {
         }
     }
 
-    /// Begin mining an epoch's transactions.
-    /// Returns an open ClarityTx for mining the block, as well as the ExecutionCost of any confirmed
-    ///  microblocks.
-    /// NOTE: even though we don't yet know the block hash, the Clarity VM ensures that a
-    /// transaction can't query information about the _current_ block (i.e. information that is not
-    /// yet known).
-    pub fn epoch_begin<'a>(
+    /// This function should be called before `epoch_begin`.
+    /// It loads the parent microblock stream, sets the parent microblock, and returns
+    /// data necessary for `epoch_begin`.
+    /// Returns chainstate transaction, clarity instance, burnchain header hash
+    /// of the burn tip, burn tip height + 1, the parent microblock stream,
+    /// the parent consensus hash, the parent header hash, and a bool
+    /// representing whether the network is mainnet or not.
+    pub fn pre_epoch_begin<'a>(
         &mut self,
         chainstate: &'a mut StacksChainState,
         burn_dbconn: &'a SortitionDBConn,
-    ) -> Result<(ClarityTx<'a>, ExecutionCost), Error> {
-        let mainnet = chainstate.config().mainnet;
-
-        // find matured miner rewards, so we can grant them within the Clarity DB tx.
-        let (latest_matured_miners, matured_miner_parent) = {
-            let mut tx = chainstate.index_tx_begin()?;
-            let latest_miners =
-                StacksChainState::get_scheduled_block_rewards(&mut tx, &self.chain_tip)?;
-            let parent_miner =
-                StacksChainState::get_parent_matured_miner(&mut tx, mainnet, &latest_miners)?;
-            (latest_miners, parent_miner)
-        };
-
-        // there's no way the miner can learn either the burn block hash or the stacks block hash,
-        // so use a sentinel hash value for each that will never occur in practice.
-        let new_consensus_hash = MINER_BLOCK_CONSENSUS_HASH.clone();
-        let new_block_hash = MINER_BLOCK_HEADER_HASH.clone();
-
+    ) -> Result<
+        (
```

Comment:
This is a pretty big tuple, which makes it hard to keep track of the individual fields.  If we end up keeping `pre_epoch_begin()`, can you put this data into a dedicated struct?